src/components/DatePicker/DatePicker.js

```diff
@@ -23,11 +23,13 @@ export default class DatePicker extends Component {
     datePickerType: PropTypes.string,
     dateFormat: PropTypes.string,
     value: PropTypes.string,
+    onChange: PropTypes.func,
   };
 
   static defaultProps = {
     short: false,
     dateFormat: 'm/d/Y',
+    onChange: () => {},
```

Comment:
Instead of allocating the empty function here, could we just make assertions on if this `prop` is available and then invoke it?

Reply:
I am personally more a fan of a default rather than the cognitive load of having to put a value check on every use case (given the sacrifice is just a single function allocation per instance). But I will follow the repo standard if there's a good case for it.

Reply:
The rationale is that we don't want to have to initialize empty closures when we don't have to 😄 

Since we are a library, performance is one of our guiding principles that we try to follow. While this is a very small thing that ultimately wouldn't be the bottleneck in an application, the hope is that we can try to stay as lean as possible on behalf of consumers.

If the empty function is indeed a pattern that we'd want to follow, then we would want to share the same empty function reference between props, similar to Facebook's [`emptyFunction`](https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/core/emptyFunction.js).